<launch>
    <arg name="namespace"           default="$(env UAV_NAMESPACE)"/>
    <arg name="input_trajectory"    default="tracker/input_trajectory"/>
    <arg name="carrot_trajectory"   default="carrot/trajectory"/>
    <arg name="carrot_status"       default="carrot/status"/>
    <arg name="output_point"        default="position_hold/trajectory"/>
    <arg name="output_pose"         default="mavros/setpoint_position/local"/>
    <arg name="odom"                default="mavros/local_position/odom"/>
    <arg name="tracker_config"      default="$(find uav_ros_tracker)/config/topp_config_marcus.yaml" />
    <arg name="tracker_path"        default="tracker/path"/>
    <arg name="carrot_pose"         default="carrot/pose"/>

    <!-- Trajectory generation -->
    <node name="topp_trajectory_gen" pkg="topp_ros" type="generate_toppra_trajectory.py" 
        output="screen" ns="$(arg namespace)"/>

    <!-- Trajectory point publisher -->
    <node name="toppra_uav_ros_tracker" pkg="uav_ros_tracker" type="topp_tracker.py" 
        ns="$(arg namespace)" output="screen">
        <remap from="tracker/input_trajectory"  to="$(arg input_trajectory)"/>
        <remap from="carrot/trajectory"         to="$(arg carrot_trajectory)"/>
        <remap from="carrot/status"             to="$(arg carrot_status)"/>
        <remap from="output/point"              to="$(arg output_point)"/>
        <remap from="output/pose"               to="$(arg output_pose)"/>
        <remap from="odometry_topic"            to="$(arg odom)"/>
        <remap from="tracker/path"              to="$(arg tracker_path)"/>
        <remap from="carrot/pose"               to="$(arg carrot_pose)"/>

        <rosparam file="$(arg tracker_config)" command="load"/>
    </node>

</launch>
